hydra-node/src/Hydra/Chain/Direct/Handlers.hs

```diff
@@ -310,14 +310,20 @@ fromPostChainTx TimeHandle{currentPointInTime, adjustPointInTime} cardanoKeys wa
     (ContestTx{confirmedSnapshot}, TkClosed) -> do
       shifted <- throwLeft $ adjustPointInTime closeGraceTime pointInTime
       pure (contest confirmedSnapshot shifted st)
-    (FanoutTx{utxo}, TkClosed) ->
-      pure (fanout utxo pointInTime st)
+    (FanoutTx{utxo}, TkClosed) -> do
+      -- NOTE: It's a bit weir that we inspect the state here, but handing
+      -- errors around while we want the possibly failing "time -> slot"
+      -- conversion to be done here is not prettier.
+      deadlineSlot <- throwLeft . slotFromPOSIXTime $ getContestationDeadline st
+      pure (fanout utxo deadlineSlot st)
```

Comment:
Remark: There's still a good chance that this is rejected by the network if submitted too early. So far this burden fall to the client, but this is where retrying would greatly improve UX.

Reply:
Yes. It's not the full solution as mentioned in the comments.